### PR TITLE
Adding error details to errors

### PR DESCRIPTION
**Pull request by balazssimongen**

### CODE REVIEW

File: src/genesys/genesys-destination-adapter.ts [hunk 1]
```diff
@@ -94,6 +94,7 @@ export class GenesysDestinationAdapter implements DestinationAdapter {
       throw new InvalidExportJobError(
         'Missing downloadURL from destination export job ' +
           JSON.stringify(job),
```
[the remark this answers is not in the document]
Comment (balazssimongen) [reply src/genesys/genesys-destination-adapter.ts:96]: I think it is unnecessary to add the whole export job object as details. I added the job id to the details, so if anyone is interested they can query it

[the remark this answers is not in the document]
Comment (kinpeter-gsys) [reply src/genesys/genesys-destination-adapter.ts:96]: Yeah, I see, but now you put the "whole export job object" stringified to the message, is that necessary? I was just wondering maybe the message could be shorter and the job details could be in the details object - at least the part of it which is useful to analyze the error.

File: src/utils/errors/error-base.ts [hunk 2]
```diff
@@ -1,13 +1,23 @@
 export abstract class ErrorBase extends Error {
   protected code: string;
+  protected details: { [key: string]: any };
 
-  public constructor(code: string, message: string) {
+  public constructor(
+    code: string,
+    details: { [key: string]: any },
```
Comment (petergaras) [inline src/utils/errors/error-base.ts:7]: I would put details after message, and make it optional.

File: src/genesys/genesys-destination-adapter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (petergaras) [reply src/genesys/genesys-destination-adapter.ts:96]: I think it is enough to log the job object, and only include the id into the message

File: src/utils/errors/error-base.ts (hunk 2 above)
Comment (petergaras) [reply src/utils/errors/error-base.ts:7]: That would also makes descendants cleaner
`
export class ValidationError extends ErrorBasePublic {
  constructor(message: string, details: { [key: string]: any }) {
    super(ErrorCodes.VALIDATION_ERROR, message, details);
  }
}
`

File: src/genesys/genesys-destination-adapter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (balazssimongen) [reply src/genesys/genesys-destination-adapter.ts]: I changed it, so now only the job id will be in the message and the job object will be in errorDetails

**Review by kinpeter-gsys — APPROVED** [review]

**Review by petergaras — APPROVED** [review]

**Pull request merged**